### ISSUE TITLE
Overlay flickers when component re-renders

### ISSUE THREAD

**Issue by Douglas-Hsieh** [Charanor/react-native-highlight-overlay#7]
Hello, thank you for this package, it's very useful :D

I found out that when highlight is active and a component containing HighlightOverlay re-renders, the overlay will "flicker"

I tried using React.memo on HighlightOverlay, because I thought the "flicker" was caused by HighlightOverlay re-rendering when the parent component re-renders.

Here is a repro (modified version of example/App.tsx)

```tsx
/* eslint-disable */
import React, { useState, useEffect, memo } from "react";
import { ScrollView, StyleSheet, View } from "react-native";
import { HighlightableElementProvider, HighlightOverlay } from "react-native-highlight-overlay";

import FavoriteList from "./components/FavoriteList";
import Header from "./components/Header";
import SimilarList from "./components/SimilarList";

export const HIGHLIGHTED_ID_1 = "one";
export const HIGHLIGHTED_ID_2 = "two";

function App() {
  const [rootRef, setRootRef] = useState<React.Component<unknown> | null>(null);
  const [highlightedId, setHighlightId] = useState<string | null>(null);
  const [counter, setCounter] = useState<number>(0);

  useEffect(() => {
    setInterval(() => {
      console.log(counter)
      setCounter(Math.random())
    }, 1000);
  }, [])

  const MemoizedHighlightOverlay = memo(HighlightOverlay)

  return (
    <HighlightableElementProvider rootRef={rootRef}>
      <Header />
      <ScrollView ref={setRootRef}>
        <View style={styles.container}>
          <FavoriteList {...{ setHighlightId }} />
          <SimilarList {...{ setHighlightId }} />
        </View>
        <MemoizedHighlightOverlay
          highlightedElementId={highlightedId}
          onDismiss={() => setHighlightId(null)}
        />
      </ScrollView>
    </HighlightableElementProvider>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#F0F0F0",
    padding: 25,
  },
});

export default App;
```

**Comment by Charanor** [comment]
Thank you! I'm sorry it took so long to get back to you but I've been very busy and haven't had time to do any work on this. Now I finally have some free time and will be addressing all of these bugs :) Thank you for the example I'll look into it!

**Comment by Charanor** [comment]
This should hopefully be fixed in the latest release 1.3.0 🎉
